tests/test_globalization.py

```diff
@@ -0,0 +1,155 @@
+# coding=utf-8
+import unittest
+import socket
+from utility import random_str, decode
+from mssqltestutils import create_mssql_cli_client, shutdown
+
+
+# All tests require a live connection to test server.
+# Make modifications to mssqlutils.create_mssql_cli_client() to use a different server and database.
+# Please Note: These tests cannot be run offline.
+class GlobalizationResultSetTests(unittest.TestCase):
+    
+    def test_charset_double(self):
+        charset_list = self.get_charset_double() 
+        self.run_charset_validation(charset_list)
+    
+
+    def test_charset_four(self):
+        charset_list = self.get_charset_four() 
+        self.run_charset_validation(charset_list)
+
+
+    def test_charset_mongolian(self):
+        charset_list = self.get_charset_mongolian()
+        self.run_charset_validation(charset_list)
+
+
+    def test_charset_uyghur(self):
+        charset_list = self.get_charset_uyghur()
+        self.run_charset_validation(charset_list)
+
+
+    def test_charset_tibetian(self):
+        charset_list = self.get_charset_tibetian()
+        self.run_charset_validation(charset_list)
+
+
+    def test_charset_yi(self):
+        charset_list = self.get_charset_yi() 
+        self.run_charset_validation(charset_list)
+
+
+    def test_charset_zang(self):
+        charset_list = self.get_charset_zang() 
+        self.run_charset_validation(charset_list)
+
+    
+    def run_charset_validation(self, chars_list):
+        """
+            Verify the column names and string values in rows returned by select statement are properly encoded as unicode.
+        """
+        local_machine_name = socket.gethostname().replace('-','_').replace('.','_')
+        try:
+            client = create_mssql_cli_client()
+            max_string_length = 50
+            for chars in chars_list:
+                while len(chars) > 0:
+                    test_string_length = min(len(chars), max_string_length)
+                    test_str = chars[0:test_string_length]
+                    chars = chars[test_string_length:]
+                    col1_name = u'col_{0}1'.format(test_str)
+                    col2_name = u'col_{0}2'.format(test_str)
+                    table_name = u'#mssqlcli_{0}_{1}_{2}'.format(local_machine_name, random_str(), test_str)
+                    setup_query = u"CREATE TABLE {0} ({1} nvarchar(MAX), {2} int);"\
+                        u"INSERT INTO {0} VALUES (N'value_{3}1', 1);"\
+                        u"INSERT INTO {0} VALUES (N'value_{3}2', 2);"\
+                        .format(table_name, col1_name, col2_name, test_str)
```

Comment:
>format [](start = 25, length = 6)

Can we insert the entire chars string?

Reply:
And validate the chars string.

---
In reply to: [324836186](https://github.com/dbcli/mssql-cli/pull/293#discussion_r324836186) [](ancestors = 324836186)

Reply:
synced offline.